test/scheduler_test.rb

```diff
@@ -585,6 +585,44 @@ def enqueue_started!
     assert_equal 1, Resque.size('quick')
   end
 
+  context 'dynamic mode is on' do
+    setup do
+      Resque::Scheduler.dynamic = true
+    end
+
+    test '#purge_dynamic_schedule! does not delete redis keys' do
+      Resque.set_schedule(
+        'some_ivar_job',
+        'cron' => '* * * * *', 'class' => 'SomeIvarJob', 'args' => '/tmp/2 ',
+        'persist' => true
+      )
+      assert_equal 1, Resque.redis.scard(:schedules_changed)
+      assert_equal 1, Resque.redis.hlen(:persistent_schedules)
+      Resque.purge_dynamic_schedule!
+      assert_equal 1, Resque.redis.scard(:schedules_changed)
+      assert_equal 1, Resque.redis.hlen(:persistent_schedules)
+    end
+  end
+
+  context 'dynamic mode is off' do
+    setup do
+      Resque::Scheduler.dynamic = false
+    end
+
+    test '#purge_dynamic_schedule! deletes redis keys' do
+      Resque.set_schedule(
+        'some_ivar_job',
+        'cron' => '* * * * *', 'class' => 'SomeIvarJob', 'args' => '/tmp/2 ',
+        'persist' => true
+      )
+      assert_equal 1, Resque.redis.scard(:schedules_changed)
+      assert_equal 1, Resque.redis.hlen(:persistent_schedules)
+      Resque.purge_dynamic_schedule!
+      assert_equal 0, Resque.redis.scard(:schedules_changed)
+      assert_equal 0, Resque.redis.hlen(:persistent_schedules)
+    end
```

Comment:
note, not testing that `reload_schedule!` is called, because it seems difficult to mock all of that out and the core part here is that redis keys are being deleted. lmk what you think.